corehq/apps/data_dictionary/models.py

```diff
@@ -166,14 +166,32 @@ def delete(self, *args, **kwargs):
         return super(CaseProperty, self).delete(*args, **kwargs)
 
     def check_validity(self, value):
-        if value and self.data_type == 'date':
-            try:
-                datetime.strptime(value, ISO_DATE_FORMAT)
-            except ValueError:
-                raise exceptions.InvalidDate(sample=value)
-        elif value and self.data_type == 'select' and self.allowed_values.exists():
-            if not self.allowed_values.filter(allowed_value=value).exists():
-                raise exceptions.InvalidSelectValue(sample=value, message=self.valid_values_message)
+        if value is None or value == '':
+            return
+        validation_methods = {
+            self.DataType.DATE.value: self._validate_date,
+            self.DataType.SELECT.value: self._validate_select,
+        }
+        method = validation_methods.get(
+            self.data_type,
+            lambda x: None
+        )
+        return method(value)
```

Comment:
I love factoring out the old code into the validation methods, but especially since our default validation is a no-op, I'm curious why we opted for a `get` approach over:
```
if self.data_type == self.DataType.DATA.value:
  self._validate_date(value)
elif self.data_type == self.DataType.SELECT.value:
  self._validate_select(value)
```
Are we anticipating many more validation methods in the future?

Also, nit: it doesn't look like the validation methods are meant to return a value, so I'm curious if the `return method(value)` statement should instead forego the `return`.

Reply:
:+1:  55977607bff